campaign-launcher/interfaceV2/src/components/FormExchangeSelect/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { Autocomplete, Box, TextField } from "@mui/material";
+import { ControllerRenderProps, FieldValues, Path } from 'react-hook-form';
+
+import { useExchangesContext } from "../../providers/ExchangesProvider";
+import { CryptoEntity } from "../CryptoEntity";
+
+type FormExchangeSelectProps<TFieldValues extends FieldValues, TName extends Path<TFieldValues>> = {
+  field: ControllerRenderProps<TFieldValues, TName>;
+  disabled?: boolean;
+}
+
+const slotProps = {
+  paper: {
+    elevation: 4,
+    sx: {
+      bgcolor: 'background.default',
+    },
+  },
+};
+
+const FormExchangeSelect = <
+  TFieldValues extends FieldValues, 
+  TName extends Path<TFieldValues>
+>({ field, disabled = false }: FormExchangeSelectProps<TFieldValues, TName>) => {
+  const { exchanges } = useExchangesContext();
+
+  return (
+    <Autocomplete
+      id="exchange-select"
+      slotProps={slotProps}
+      options={exchanges?.map((exchange) => exchange.name) || []}
+      getOptionLabel={(option) => {
+        const exchange = exchanges?.find((e) => e.name === option);
```

Comment:
The list of exchanges is pretty big, so maybe we optimize it somehow?
E.g. context to provide a `Map<exchangeName, exchangeData>` and work with this map?